### PR TITLE
WIP: Draw dispatch review branch.

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
code review of draw dispatch additions to syval phase1 staging

### CODE REVIEW

File: layers/synchronization_validation.cpp [hunk 1]
```diff
@@ -2315,3 +2342,1208 @@ void SyncValidator::PreCallRecordCmdBlitImage(VkCommandBuffer commandBuffer, VkI
...
+                if (descriptor->GetClass() == DescriptorClass::ImageSampler || descriptor->GetClass() == DescriptorClass::Image) {
+                    const IMAGE_VIEW_STATE *img_view_state = nullptr;
+                    if (descriptor->GetClass() == DescriptorClass::ImageSampler) {
+                        img_view_state = static_cast<const ImageSamplerDescriptor *>(descriptor)->GetImageViewState();
+                    } else {
+                        img_view_state = static_cast<const ImageDescriptor *>(descriptor)->GetImageViewState();
+                    }
+                    if (!img_view_state) continue;
+                    SyncStageAccessIndex sync_index;
+                    if (descriptor_type == VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT) {
```
Comment (jzulauf-lunarg) [inline layers/synchronization_validation.cpp:2394]: I don't think you're covering all the usages.  VK_IMAGE_USAGE_SAMPLED_BIT  for example -- can be a sampled image with either FRAGMENT or VERTEX shader read, no?

File: layers/synchronization_validation.cpp [hunk 2]
```diff
@@ -102,6 +102,33 @@ static constexpr SyncOrderingBarrier kAttachmentRasterOrder = {kDepthStencilAtta
...
+    VK_ACCESS_INDIRECT_COMMAND_READ_BIT | VK_ACCESS_INDEX_READ_BIT | VK_ACCESS_VERTEX_ATTRIBUTE_READ_BIT |
+    VK_ACCESS_UNIFORM_READ_BIT | VK_ACCESS_INPUT_ATTACHMENT_READ_BIT | VK_ACCESS_SHADER_READ_BIT |
+    VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_READ_BIT | VK_ACCESS_TRANSFER_READ_BIT |
+    VK_ACCESS_HOST_READ_BIT | VK_ACCESS_MEMORY_READ_BIT | VK_ACCESS_TRANSFORM_FEEDBACK_COUNTER_READ_BIT_EXT |
+    VK_ACCESS_CONDITIONAL_RENDERING_READ_BIT_EXT | VK_ACCESS_COLOR_ATTACHMENT_READ_NONCOHERENT_BIT_EXT |
+    VK_ACCESS_ACCELERATION_STRUCTURE_READ_BIT_KHR | VK_ACCESS_SHADING_RATE_IMAGE_READ_BIT_NV |
+    VK_ACCESS_FRAGMENT_DENSITY_MAP_READ_BIT_EXT | VK_ACCESS_COMMAND_PREPROCESS_READ_BIT_NV |
+    VK_ACCESS_ACCELERATION_STRUCTURE_READ_BIT_NV;
+
+static constexpr VkAccessFlags ALLWriteAccessMask =
```
[the remark this answers is not in the document]
Comment (locke-lunarg) [reply layers/synchronization_validation.cpp]: Removed it. Added MEMORY_READ/WRITE in sync_val_gen.py.

File: layers/synchronization_validation.cpp [hunk 3]
```diff
@@ -102,6 +102,33 @@ static constexpr SyncOrderingBarrier kAttachmentRasterOrder = {kDepthStencilAtta
...
+    VK_ACCESS_SHADER_WRITE_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT | VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT |
+    VK_ACCESS_TRANSFER_WRITE_BIT | VK_ACCESS_HOST_WRITE_BIT | VK_ACCESS_MEMORY_WRITE_BIT |
+    VK_ACCESS_TRANSFORM_FEEDBACK_WRITE_BIT_EXT | VK_ACCESS_TRANSFORM_FEEDBACK_COUNTER_WRITE_BIT_EXT |
+    VK_ACCESS_ACCELERATION_STRUCTURE_WRITE_BIT_KHR | VK_ACCESS_COMMAND_PREPROCESS_WRITE_BIT_NV |
+    VK_ACCESS_ACCELERATION_STRUCTURE_WRITE_BIT_NV;
+
+inline VkAccessFlags GetMemoryAccessMask(VkAccessFlags accessMask) {
```
[the remark this answers is not in the document]
Comment (locke-lunarg) [reply layers/synchronization_validation.cpp]: Removed it. Added MEMORY_READ/WRITE in sync_val_gen.py.

File: layers/synchronization_validation.cpp [hunk 4]
```diff
@@ -845,7 +872,7 @@ SyncStageAccessFlags SyncStageAccess::AccessScopeByStage(VkPipelineStageFlags st
 }
 
 SyncStageAccessFlags SyncStageAccess::AccessScopeByAccess(VkAccessFlags accesses) {
-    return AccessScopeImpl(accesses, syncStageAccessMaskByAccessBit);
+    return AccessScopeImpl(GetMemoryAccessMask(accesses), syncStageAccessMaskByAccessBit);
```
[the remark this answers is not in the document]
Comment (locke-lunarg) [reply layers/synchronization_validation.cpp]: Added MEMORY_READ/WRITE in sync_val_gen.py.

File: layers/synchronization_validation.cpp [hunk 5]
```diff
@@ -2315,3 +2342,1208 @@ void SyncValidator::PreCallRecordCmdBlitImage(VkCommandBuffer commandBuffer, VkI
...
+    using DescriptorClass = cvdescriptorset::DescriptorClass;
+    using BufferDescriptor = cvdescriptorset::BufferDescriptor;
+    using ImageDescriptor = cvdescriptorset::ImageDescriptor;
+    using ImageSamplerDescriptor = cvdescriptorset::ImageSamplerDescriptor;
+    using TexelDescriptor = cvdescriptorset::TexelDescriptor;
+
+    for (const auto &set_binding_pair : pPipe->active_slots) {
+        uint32_t setIndex = set_binding_pair.first;
+        cvdescriptorset::DescriptorSet *descriptor_set = state.per_set[setIndex].bound_descriptor_set;
+        cvdescriptorset::PrefilterBindRequestMap reduced_map(*descriptor_set, set_binding_pair.second);
```
[the remark this answers is not in the document]
Comment (locke-lunarg) [reply layers/synchronization_validation.cpp]: Removed it.

File: layers/synchronization_validation.cpp (hunk 1 above)
Comment (locke-lunarg) [reply layers/synchronization_validation.cpp]: @jzulauf-lunarg  Detect/Update every shader stages has been done. Writable still WIP.

**Pull request closed without merging**